### ISSUE TITLE
How to check signature when the input is Windows physical drive device?

### ISSUE THREAD

**Issue by Glimpse-006** [libyal/libvslvm#8]
```
iResult =lvm::libvslvm_check_volume_signature("c:\\aa.img", &pError);
if (lvm_result_succeed != iResult)
{
	lvm::libvslvm_error_sprint(pError, pchErrorInfo, dwErrorSize);			
}
 itis right, but 
iResult =lvm::libvslvm_check_volume_signature("\\\\.\\physicaldrive1", &pError);
error info:libvslvm_check_volume_signature: unable to set filename in file IO handle."	
```

**Comment by joachimmetz** [comment]
I have a hard time trying to parse what you are trying to tell me, could you please be more explanatory.

A Windows physical drive device does not behave like a "regular" file. You can try using libmsdev or create your own libbfio (file IO) handle to handle the Windows physical drive device behavior.

**Comment by Glimpse-006** [comment]
I downloaded and compiled your library: libbfio, modify the code as follows: but reported memory error.(It's translated using tools -_-)
```
HANDLE hFile = CreateFile(L"\\\\.\\PhysicalDrive0", GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_WRITE,  NULL, OPEN_EXISTING, NULL, NULL);
if (hFile == INVALID_HANDLE_VALUE)
{
	DWORD dwError;
	dwError = GetLastError();
	return -1;
}
lvm::libbfio_handle_t *p_bfio_handle = NULL;
HANDLE hDeviceTemp = NULL;

iResult = lvm::libbfio_handle_initialize(&p_bfio_handle, 
	(intptr_t*)hFile,
	&lvm::libbfio_handle_free,
	&lvm::libbfio_handle_clone,
	&lvm::libbfio_handle_open,
	&lvm::libbfio_handle_close,
	&lvm::libbfio_handle_read_buffer,
	&lvm::libbfio_handle_write_buffer,
	&lvm::libbfio_handle_seek_offset,
	&lvm::libbfio_handle_exists,
	&lvm::libbfio_handle_is_open,
	&lvm::libbfio_handle_get_size,
	lvm::LIBVSLVM_ACCESS_FLAG_READ, 
	&pError);
iResult = lvm::libvslvm_check_volume_signature_file_io_handle(p_bfio_handle, &pError);
if (lvm_result_succeed != iResult)
{
	lvm::libvslvm_error_sprint(pError, pchErrorInfo, dwErrorSize);
	break;
}
```

**Comment by joachimmetz** [comment]
@Glimpse-006 I have to little context to give you any advise on "memory errors". I realize that the language barrier is not helping, could please explain step-by-step in as much detail as possible what you are doing and what is not working. Please add error messages and other relevant details.

**Comment by Glimpse-006** [comment]
I want to use your library to get LVM information on a physical disk, or on a volume.
I don't know whether I have used your code correctly. According to my understanding, I just wrote it like this.

Use the code above ,debug  the function "libvslvm_check_volume_signature_file_io_handle" step by step, when  goto the position 1612 of the document" libbfio_handle.c", the content is 
if(internal_handle->io_handle == NULL),
Throws an exception: read access conflict, handle is 0x34( the value of hFile is 0x34)
